### PR TITLE
AKS - Azure CNI and Calico Network Policy (#48)

### PR DESCRIPTION
**Pull request by Adeelku**
## Overview/Summary

Adding Option for the creation of AKS Cluster with one of the following three scenarios:

* Network Plugin: Kubenet + Network Policy: Calico (Network Policy)
* Network Plugin: Azure CNI + Network Policy: Calico (Network Policy)
* Network Plugin: Azure CNI + Network Policy: Azure (Network Policy)

Kubenet + Calico

image

Azure CNI + Calico

image

Azure CNI + Azure NP

image

## This PR fixes/adds/changes/removes

1. *Renames the aks-kubenet module => aks*
2. *Updates the AKS section in the parameters file*
````json
    "aks": {
      "value": {
        "version": "1.21.2",
        "networkPlugin": "kubenet" ,
        "networkPolicy": "calico",
        "podCidr": "11.0.0.0/16",
        "serviceCidr": "20.0.0.0/16" ,
        "dnsServiceIP": "20.0.0.10",
        "dockerBridgeCidr": "30.0.0.1/16"
      }
    },
````
3. *Updates Doc: machinelearning.md*
4. *Update /schemas/latest/readme.md*

### Breaking Changes
New schema version and test cases added. Breaking old schema.

## Testing Evidence

pwsh -File ./test-all.ps1 -TestFolder . -SchemaFolder ../../schemas/latest/landingzones ==> succeeded

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/azure/CanadaPubSecALZ/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/azure/CanadaPubSecALZ/issues)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/azure/CanadaPubSecALZ/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

File: azresources/containers/aks/aks-with-cmk.bicep [hunk 1]
```diff
@@ -86,6 +92,18 @@ param akvResourceGroupName string
 @description('Azure Key Vault Name.  Required when useCMK=true.')
 param akvName string
 
+var podCidra = contains(networkPlugin, 'azure') ? null : podCidr
```
Comment (SenthuranSivananthan) [inline azresources/containers/aks/aks-with-cmk.bicep:95]: What does "a" in `podCidra` mean?  Can you write out the variable in full so it's self describing?

File: azresources/containers/aks/aks-without-cmk.bicep [hunk 2]
```diff
@@ -79,7 +85,19 @@ param containerInsightsLogAnalyticsResourceId string = ''
 @description('Enable encryption at host (double encryption).  Default: true')
 param enableEncryptionAtHost bool = true
 
-resource akskubenet 'Microsoft.ContainerService/managedClusters@2021-07-01' = {
+var podCidra = contains(networkPlugin, 'azure') ? null : podCidr
```
Comment (SenthuranSivananthan) [inline azresources/containers/aks/aks-without-cmk.bicep:88]: What does "a" in `podCidra` mean? Can you write out the variable in full so it's self describing?

File: config/variables/adiltouati-main.yml [hunk 3]
```diff
@@ -0,0 +1,285 @@
+# ----------------------------------------------------------------------------------
```
Comment (SenthuranSivananthan) [inline config/variables/adiltouati-main.yml]: Can you remove your environment config?

File: config/variables/common.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@
 variables:
 
   deploymentRegion: canadacentral
-  serviceConnection: spn-azure-platform-ops
```
Comment (SenthuranSivananthan) [inline config/variables/common.yml:13]: Can you reset this variable to the original version?

```
serviceConnection: spn-azure-platform-ops
```

File: schemas/latest/landingzones/types/aks.json [hunk 5]
```diff
@@ -15,17 +15,26 @@
...
-              "version": {
-                  "type": "string"
-              }
-          },
-          "required": [
-              "version"
-          ],
-          "title": "Value"
-      }
+        "type": "object",
```
Comment (SenthuranSivananthan) [inline schemas/latest/landingzones/types/aks.json:18]: Can you update the changelog located in `schemas/latest/readme.md`

**Comment by Adeelku** [comment]
@SenthuranSivananthan ,I updated /schemas/latest/readme.md

File: schemas/latest/landingzones/types/aks.json (hunk 5 above)
Comment (SenthuranSivananthan) [reply schemas/latest/landingzones/types/aks.json:18]: Can you also parameterize these parameters?  They are hardcoded as defaults and would be better to move them out to config as part of this change.

- podCidr // default = '11.0.0.0/16'
- serviceCidr // default = '20.0.0.0/16'
- dnsServiceIP // default = '20.0.0.10'
- dockerBridgeCidr // default = '30.0.0.1/16'

**Review by SenthuranSivananthan — COMMENTED** [review]
Thank you for this PR @Adeelku.  I've left a few comments that I hope you can update.

I'm wondering whether you can parameterize the AKS IP addresses such as podCidr, serviceCidr, etc. so that the `aks` configuration block in our parameters is more comprehensive?

File: config/variables/adiltouati-main.yml (hunk 3 above)
Comment (Adeelku) [reply config/variables/adiltouati-main.yml]: Sure, I believe I did it in the new update of the PR, but will do

File: config/variables/common.yml (hunk 4 above)
Comment (Adeelku) [reply config/variables/common.yml]: Ready for review!

**Review by SenthuranSivananthan — COMMENTED** [review]
Can you update the deployment scenarios table in `docs/archetypes/machinelearning.md`?  You should also update the example parameters JSON in the doc so that it includes the updated information.

I think we are good to merge after this change.

**Comment by Adeelku** [comment]
@SenthuranSivananthan  docs/archetypes/machinelearning.md updated!

**Review by SenthuranSivananthan — APPROVED** [review]
Thank you @Adeelku.  PR approved.

**Pull request merged**